plugins/modules/sonarr_quality_profile.py

```diff
@@ -230,58 +196,67 @@ def run_module():
+            quality_groups_dict[item.quality.name] = item
+
     # Populate quality groups.
-    quality_groups = []
-    for item in module.params['quality_groups']:
+    ident = 1000
+    cutoff_id = None
+    for item in reversed(module.params['quality_groups']):
+        name = item['qualities'][0]
         if len(item['qualities']) == 1:
-            quality_groups.append(sonarr.QualityProfileQualityItemResource(**{
-                'quality': sonarr.Quality(**{
-                    'id': item['qualities'][0]['id'],
-                    'name': item['qualities'][0]['name'],
-                    'source': item['qualities'][0]['source'],
-                    'resolution': item['qualities'][0]['resolution'],
-                }),
-                'items': [],
-                'allowed': True,
-            }))
+            quality_groups_dict[name].allowed = True
+            if name == module.params['cutoff']:
+                cutoff_id = quality_groups_dict[name].quality.id
         else:
-            qualities = []
-            for quality in item['qualities']:
-                qualities.append(sonarr.QualityProfileQualityItemResource(**{
-                    'quality': sonarr.Quality(**{
-                        'id': quality['id'],
-                        'name': quality['name'],
-                        'source': quality['source'],
-                        'resolution': quality['resolution'],
-                    }),
-                    'allowed': True,
-                    'items': []
-                }))
-
-            quality_groups.append(sonarr.QualityProfileQualityItemResource(**{
+            qualities = [quality_groups_dict[name].copy()]
+            quality_groups_dict[name].allowed = True
+            for quality in item['qualities'][1:]:
+                qualities.append(quality_groups_dict.pop(quality))
+                qualities[-1].allowed = True
+
+            quality_groups_dict[name] = quality_groups_dict[name].copy(update={
                 'allowed': True,
                 'name': item['name'],
-                'id': item['id'],
+                'id': ident,
+                'quality': None,
                 'items': qualities,
-            }))
+            })
+
+            if item['name'] == module.params['cutoff']:
+                cutoff_id = ident
+            ident += 1
+        quality_groups_dict.move_to_end(name, last=True)
+    want.items = list(quality_groups_dict.values())
 
     # Populate formats.
-    formats = []
-    for item in module.params['formats']:
-        formats.append(sonarr.ProfileFormatItemResource(**{
-            'name': item['name'],
-            'format': item['id'],
-            'score': item['score'],
-        }))
-
-    want = sonarr.QualityProfileResource(**{
+    formats_dict = {item['name']: item for item in want.format_items}
+    for key, value in module.params['formats'].items():
+        formats_dict[key].score = value
+    want.format_items = list(formats_dict.values())
```

Comment:
I don't get this, let's say I want to specify different scores to different custom formats, am I able to do that?
However, I like the idea of populating the formats from here. Right now there is probably an issue if you add a custom format but you don't update your quality profile with the new custom format. On the first edit it'll surely fail.

Reply:
Yea, custom formats are now just speicifed as
```yaml
formats:
  AAC: 1000
  DTS: 2000
```
Or something like that. If you add a custom format and don't update the quality profile nothing should break though. It'll grab the new schema, apply the modifications, see that the result is the same as you get in Sonarr(when a new custom format is added it's added to each quality profile with a score of zero) and everything will be fine?